src/shogun/lib/DynamicArray.h

```diff
@@ -537,11 +537,15 @@ template <class T> class CDynamicArray :public CSGObject
 			return *this;
 		}
 
-		/** shuffles the array (not thread safe!) */
-		inline void shuffle() { m_array.shuffle(); }
+		inline auto begin()
```

Comment:
For this to work with swig you need to add `->decltype(m_array.end())` as trailing return type. Swig is C++11 compatible and just writing auto is C++14 (I think)

Reply:
And if that doesn't work you can just explicitly write RandomIterator<T>, or even hide it from SWIG, since this shouldn't be available from the interface anyway

Reply:
I looked up iterators at SGVector and found that they are removed from swig "#ifndef SWIG", so I did the same.

Reply:
yup that makes sense!